File: ComPDFKit for Windows/Examples/Samples/TextExtractTest/VB/TextExtractTest.vb
Imports ComPDFKit.PDFDocument
Imports System
Imports System.IO

Module TextExtractTest
    Private parentPath As String =
        Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(Directory.GetCurrentDirectory())))
    Private outputPath As String = Path.Combine(parentPath, "Output", "VB")

    Sub Main(args As String())
        ' Preparation work
        Console.WriteLine("Running PDFPage test sample..." & vbCrLf)

        SDKLicenseHelper.LicenseVerify()
        Dim document As CPDFDocument = CPDFDocument.InitWithFilePath("CommonFivePage.pdf")

        If Not Directory.Exists(outputPath) Then
            Directory.CreateDirectory(outputPath)
        End If

        If PDFToText(document) Then
            Console.WriteLine("PDF to text done.")
        Else
            Console.WriteLine("PDF to text failed.")
        End If

        Console.WriteLine("--------------------")
        Console.WriteLine("Done!")
        Console.WriteLine("--------------------")
        Console.ReadLine()
    End Sub

    ' Convert PDF to text
    Private Function PDFToText(document As CPDFDocument) As Boolean
        Dim filePath As String = Path.Combine(outputPath, "PDFToText.txt")
        If Not document.PdfToText("1-" & document.PageCount.ToString(), filePath) Then ' Page ranges are counted from 1
            Return False
        End If
        Console.WriteLine("Browse the generated file in " & filePath)
        Return True
    End Function
End Module

File: ComPDFKit for Windows/Examples/Samples/TextExtractTest/CS/TextExtractTest.cs
using ComPDFKit.PDFDocument;
using System;
using System.IO;

namespace TextExtractTest
{
    internal class TextExtractTest
    {
        private static string parentPath =
            Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(Directory.GetCurrentDirectory())));
        private static string outputPath = Path.Combine(parentPath, "Output", "CS");
        static void Main(string[] args)
        {
            #region Perparation work
            Console.WriteLine("Running PDFPage test sample…\r\n");

            SDKLicenseHelper.LicenseVerify();
            CPDFDocument document = CPDFDocument.InitWithFilePath("CommonFivePage.pdf");

            if (!Directory.Exists(outputPath))
            {
                Directory.CreateDirectory(outputPath);
            }
            #endregion

            if (PDFToText(document))
            {
                Console.WriteLine("PDF to text done.");
            }
            else
            {
                Console.WriteLine("PDF to text failed.");
            }
            Console.WriteLine("--------------------");
            Console.WriteLine("Done!");
            Console.WriteLine("--------------------");
            Console.ReadLine();
        }

        //
        static private bool PDFToText(CPDFDocument document)
        {
            string path = Path.Combine(outputPath, "PDFToText.txt");
            if (!document.PdfToText("1-" + document.PageCount.ToString(), path))//Page ranges are counted from 1
            {
                return false;
            }
            Console.WriteLine("Browse the generated file in " + path);
            return true;
        }
    }
}
